jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoostClassifier.scala

```diff
@@ -313,7 +315,79 @@ class XGBoostClassificationModel private[ml](
 
     bBooster.unpersist(blocking = false)
 
-    dataset.sparkSession.createDataFrame(rdd, schema)
+    dataset.sparkSession.createDataFrame(rdd, generateResultSchema(schema))
+  }
+
+  private def produceResultIterator(
+      originalRowItr: Iterator[Row],
+      rawPredictionItr: Iterator[Row],
+      probabilityItr: Iterator[Row],
+      predLeafItr: Iterator[Row],
+      predContribItr: Iterator[Row]): Iterator[Row] = {
+    // the following implementation is to be improved
+    if (isDefined(leafPredictionCol) && isDefined(contribPredictionCol)) {
```

Comment:
I have a little concern that users may set ```leafPredictionCol``` with ```""```, their intention should be not to output this column, but here we don't handle this case. At MLlib, we check ```$(leafPredictionCol).isEmpty``` as well. What do you think about it?

Reply:
yes, good point, let me revise it